### PR TITLE
Fix #9810 - `format` all the things

### LINKED ISSUE #9810: `format` all the things
Issue overview
--------------

Avoid string concatenation when calling ShowXXXMessage / Error.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #9811
- I did not touch the ShowRecurringXXX functions (the parsing of their signature is much harder)

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/UtilityRoutines.cc [hunk 1]
```diff
@@ -1206,26 +1206,33 @@ void ShowContinueErrorTimeStamp(EnergyPlusData &state, std::string const &Messag
...
+                                    CreateSysTimeIntervalString(state));
+        ShowErrorMessage(state, format(" **   ~~~   ** {}", Message));
+        ShowErrorMessage(state, format(" **   ~~~   ** {}", postfix), OutUnit1, OutUnit2);
         if (state.dataSQLiteProcedures->sqlite) {
-            state.dataSQLiteProcedures->sqlite->updateSQLiteErrorRecord(m);
+            state.dataSQLiteProcedures->sqlite->updateSQLiteErrorRecord(Message);
         }
         if (state.dataGlobal->errorCallback) {
-            state.dataGlobal->errorCallback(Error::Continue, m);
+            state.dataGlobal->errorCallback(Error::Continue, Message);
```
Comment (jmarrec) [inline src/EnergyPlus/UtilityRoutines.cc:1235]: I am changing this one. Currently it's writing to the SQL (and errorCallback) with the  `** ~~~ **` in it which isn't the case in the other ladder of the if block and is not wanted as far as I can tell

File: third_party/ObjexxFCL/src/ObjexxFCL/Optional.hh [hunk 2]
```diff
@@ -265,6 +265,13 @@ present( Optional< T > const & o )
 	return o.present();
 }
 
+template <typename T>
+std::ostream& operator<< (std::ostream & os, const Optional<T>& rhs) {
+	os << T{rhs};
+	return os;
+}
```
Comment (jmarrec) [inline third_party/ObjexxFCL/src/ObjexxFCL/Optional.hh:272]: Make Optional<T> streamable.

File: src/EnergyPlus/Boilers.cc [hunk 3]
```diff
@@ -107,7 +107,7 @@ PlantComponent *BoilerSpecs::factory(EnergyPlusData &state, std::string const &o
         }
     }
     // If we didn't find it, fatal
-    ShowFatalError(state, "LocalBoilerFactory: Error getting inputs for boiler named: " + objectName); // LCOV_EXCL_LINE
+    ShowFatalError(state, format("LocalBoilerFactory: Error getting inputs for boiler named: {}", objectName)); // LCOV_EXCL_LINE
```
Comment (jmarrec) [inline src/EnergyPlus/Boilers.cc:110]: One example.

**Comment by amirroth** [comment]
🔥🔥🔥🔥🔥

**Comment by Myoldmopar** [comment]
That's an odd unit test failure given the changes here.  I am able to verify it locally, but I haven't investigated why it happens.

**Comment by jmarrec** [comment]
> That's an odd unit test failure given the changes here. I am able to verify it locally, but I haven't investigated why it happens.

fixed in https://github.com/NREL/EnergyPlus/pull/9811/commits/72a462afa73dca043743f65c787fb0af1d25d78b

It's a case of the replace code misfiring when there are old Fortran comments with ShowXXXError in there...

**Comment by Myoldmopar** [comment]
Uh oh....it's 100% clean so far, just waiting on the integration coverage to finish.  Does this just merge right in after that?  Anyone have any reason not to?  I understand this could cause conflicts, there will be major ones in my curve refactor branch, but I am happy to take a little time helping to integrate the changes in your branches, just let me know.

Anyway, speak up soon, or this may just drop right in.

**Comment by Myoldmopar** [comment]
OK, all clean.  Merging this in.  Thanks @jmarrec, now just need to add a custom_check to verify none slip in.  

As I said before, I know this will likely cause conflicts in folks branches, so just let me know if you need help.  In most cases, however, it should be a very simple conflict to resolve -- just use Julien's "format"-ed block, and then apply whatever tweak you made to the string.

**Pull request merged**